lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -182,11 +202,28 @@ def flush(&block)
 
   def close()
     cleaner.stop
+    auto_flusher.stop
     all_codecs.each(&:close)
   end
   # end Codec API
   # ==============================================
 
+  def auto_flush_mapped
+    nowf = Time.now.to_f
+    identity_map.each do |identity, compo|
+      next unless nowf > compo.auto_flush_timeout
+      compo.codec.auto_flush
```

Comment:
IIRC the Float compare drops into Java very quickly, yes it is for speed.
However if I wanted to compare Time with Time then CodecValue would need to store Time instances - consuming more memory, i.e. with thousands of identities we have 2 times that in Time instances.
We can't compare Time with Float directly.

So I choose Float or Integer most of the time.  I would use Concurrent.monotonic_time or System.nanoTime but Time.now is faster. wtf? JRuby must hold a pre-built Time instance in memory to copy from.
```sh
irb(main):016:0> Benchmark.ips do |x|
irb(main):017:1*   x.report("nano")      { java.lang.System.nanoTime() }
irb(main):018:1>   x.report("Time")      { Time.now.to_f }
irb(main):019:1>
irb(main):020:1*   x.compare!
irb(main):021:1> end
Calculating -------------------------------------
                nano   115.540k i/100ms
                Time   164.365k i/100ms
-------------------------------------------------
                nano      2.024M (± 2.6%) i/s -     10.168M
                Time      3.679M (± 2.8%) i/s -     18.409M

Comparison:
                Time:  3678931.5 i/s
                nano:  2024003.1 i/s - 1.82x slower
```